extensions/amazon-lambda/runtime/src/main/java/io/quarkus/amazon/lambda/runtime/AmazonLambdaRecorder.java

```diff
@@ -53,12 +55,17 @@ static void initializeHandlerClass(Class<? extends RequestHandler<?, ?>> handler
         ObjectMapper objectMapper = AmazonLambdaMapperRecorder.objectMapper;
         Method handlerMethod = discoverHandlerMethod(handlerClass);
         Class<?> parameterType = handlerMethod.getParameterTypes()[0];
+        Type genericParameterType = handlerMethod.getGenericParameterTypes()[0];
+        JavaType constructParameterType = objectMapper.getTypeFactory().constructType(genericParameterType);
+        Type genericReturnType = handlerMethod.getGenericReturnType();
+        JavaType constructReturnType = objectMapper.getTypeFactory().constructType(genericReturnType);
+
```

Comment:
I think you'd rather move these lines just where they are needed. As typically in the `S3Event` case, some of this is useless.